models/SignupForm.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace app\models;
+
+
+use app\helpers\MailHelper;
+use Yii;
+use yii\base\Exception;
+use yii\base\Model;
+
+/**
+ * Class SignupForm
+ *
+ * @package app\models
+ */
+class SignupForm extends Model
+{
+    public $email;
+    public $firstname;
+    public $lastname;
+    public $password;
+    public $password_repeat;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function rules()
+    {
+        return [
+            [['firstname', 'lastname'], 'string', 'max' => 255],
+            [['password'], 'string', 'min' => 6],
+            ['email', 'filter', 'filter' => 'trim'],
+            [['email', 'firstname', 'lastname'], 'required'],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function attributeLabels()
+    {
+        return [
+            'username' => Yii::t('app', 'Username'),
+            'email' => Yii::t('app', 'E-mail'),
+            'firstname' => Yii::t('app', 'Firstname'),
+            'lastname' => Yii::t('app', 'Lastname'),
+            'Password' => Yii::t('app', 'Password'),
+        ];
+    }
+
+    /**
+     * Signs user up.
+     *
+     * @param Invitation $invitation
+     * @return User|null the saved model or null if saving fails
+     * @throws Exception
+     * @throws \yii\db\Exception
+     */
+    public function signup(Invitation $invitation)
+    {
+        $dbTransaction = Yii::$app->db->beginTransaction();
+        // Save user
+        $user = new User();
+        $user->username = $this->email;
+        $user->email = $this->email;
+        $user->status = User::STATUS_INACTIVE;
+        $user->setPassword($this->password);
+        if (!$user->save()) {
+            $dbTransaction->rollBack();
+            throw new Exception("User was not saved for email $this->email");
```

Comment:
@skavt 
First of all, it is not a good idea to throw general exception. Always try to throw more specific (ValidationException etc.).
Secondary, it is better not to throw exception in this situation, because it returns entire exception as response, thus body does not contain only error string or errors object. Therefore, use `addError` (`addErrors` if multiple) method and return false from respective function. In `UserController` you have following code `$this->validationError($model->getFirstErrors())`, which will capture the error you have just added and return it correctly. 
Last but not least, `"User was not saved for email $this->email"` this message should be in `Yii::t`

Use same logic on other exception throws in this function below.